packages/inngest/src/components/connect/index.ts

```diff
@@ -1069,49 +1176,60 @@ class WebSocketWorkerConnection implements WorkerConnection {
-    const heartbeatInterval = setInterval(() => {
-      this.debug("Sending worker heartbeat", {
-        connectionId,
-      });
-
-      // Send worker heartbeat
-      ws.send(
-        ConnectMessage.encode(
-          ConnectMessage.create({
-            kind: GatewayMessageType.WORKER_HEARTBEAT,
-          })
-        ).finish()
-      );
-
-      // Wait for gateway to respond
-      setTimeout(() => {
-        if (!conn.lastGatewayHeartbeatAt) {
-          this.debug("Gateway heartbeat missed");
-          void onConnectionError(
-            new ReconnectError(
-              `Gateway heartbeat missed (${connectionId})`,
-              attempt
-            )
-          );
-          return;
-        }
-        const timeSinceLastHeartbeat =
-          new Date().getTime() - conn.lastGatewayHeartbeatAt.getTime();
-        if (timeSinceLastHeartbeat > WorkerHeartbeatInterval * 2) {
-          this.debug("Gateway heartbeat missed");
-          void onConnectionError(
-            new ReconnectError(
-              `Gateway heartbeat missed (${connectionId})`,
-              attempt
-            )
-          );
+    let heartbeatInterval = undefined;
+    if (heartbeatIntervalMs !== undefined) {
+      heartbeatInterval = setInterval(() => {
+        if (heartbeatIntervalMs === undefined) {
           return;
         }
 
-        // eslint-disable-next-line @typescript-eslint/no-floating-promises
-        this.messageBuffer.flush(hashedSigningKey);
-      }, WorkerHeartbeatInterval / 2);
-    }, WorkerHeartbeatInterval);
+        this.debug("Sending worker heartbeat", {
+          connectionId,
+        });
+
+        // Send worker heartbeat
+        ws.send(
+          ConnectMessage.encode(
+            ConnectMessage.create({
+              kind: GatewayMessageType.WORKER_HEARTBEAT,
+            })
+          ).finish()
+        );
+
+        // Wait for gateway to respond
+        setTimeout(() => {
+          if (heartbeatIntervalMs === undefined) {
+            return;
+          }
+
+          if (!conn.lastGatewayHeartbeatAt) {
+            this.debug("Gateway heartbeat missed");
+            void onConnectionError(
+              new ReconnectError(
+                `Gateway heartbeat missed (${connectionId})`,
+                attempt
+              )
+            );
+            return;
+          }
+          const timeSinceLastHeartbeat =
+            new Date().getTime() - conn.lastGatewayHeartbeatAt.getTime();
+          if (timeSinceLastHeartbeat > heartbeatIntervalMs * 2) {
+            this.debug("Gateway heartbeat missed");
+            void onConnectionError(
+              new ReconnectError(
+                `Gateway heartbeat missed (${connectionId})`,
+                attempt
+              )
+            );
+            return;
+          }
+
+          // eslint-disable-next-line @typescript-eslint/no-floating-promises
+          this.messageBuffer.flush(hashedSigningKey);
+        }, heartbeatIntervalMs / 2);
```

Comment:
A nit not about this change, but should this be something like `heartbeatIntervalMs * .9`? atm this gives the server only half the time to respond.

To be a bit more succinct, maybe this code could send a heartbeat on the interval and then check if the _previous_ heartbeat was responded to, rather than waiting for the current one with a `setTimeout`.

Reply:
Fixed this, should be cleaner now